streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -1225,8 +1225,15 @@
      * @return a {@link KStream} containing all records of the given streams
      */
     public <K, V> KStream<K, V> merge(final KStream<K, V>... streams) {
+        if (streams.length <= 1) {
```

Comment:
We also need `null` check -- one could call `stream.merge()` resulting in a NPE otherwise. Also update exception message. Just use the same as in https://github.com/apache/kafka/pull/3916/files#diff-5142e1d4a6410459d6bf6df98828e5afL352